helix-core/src/main/java/org/apache/helix/manager/zk/CallbackHandler.java

```diff
@@ -555,7 +569,11 @@ private void subscribeDataChange(String path, NotificationContext.Type callbackT
         logger.debug(_manager.getInstanceName() + " subscribe data-change. path: " + path
             + ", listener: " + _listener);
       }
-      _zkClient.subscribeDataChanges(path, this);
+      boolean rt = _zkClient.subscribeDataChanges(path, this, callbackType != Type.INIT);
+      logger.debug("CallbackHandler {} subscribe data path {} result {}", this, path, rt);
+      if (!rt) {
+        logger.info("CallbackHandler {} subscribe data path {} failed!", this, path);
```

Comment:
I am wondering if info or warn level for this message. If it may potentially cause problems maybe warn is expected. But if just regular message, info is good enough.

Reply:
As I commented before, it is ok to be returning false here.